### PR TITLE
[docs] ray.remote(object_store_memory) is for actors only

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

We explicitely disallow scheduling tasks based on object store memory, so we should state that in the docs

cc @scottsun94 

```
>>> import ray
>>> @ray.remote(object_store_memory=100)
... def foo():
...  pass
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alex/anyscale/ray/python/ray/worker.py", line 2479, in _make_remote
    ray_option_utils.validate_task_options(options, in_options=False)
  File "/Users/alex/anyscale/ray/python/ray/_private/ray_option_utils.py", line 191, in validate_task_options
    task_options[k].validate(k, v)
  File "/Users/alex/anyscale/ray/python/ray/_private/ray_option_utils.py", line 33, in validate
    raise ValueError(self.error_message_for_value_constraint)
ValueError: Setting 'object_store_memory' is not implemented for tasks
```

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/worker.py [hunk 1]
```diff
@@ -31,64 +31,54 @@
     overload,
 )
 
+import colorama
```
Comment (wuisawesome) [inline python/ray/worker.py:34]: pre-push hooks wanted this sorting.

**Comment by jjyao** [comment]
@wuisawesome For 2.0, we actually decided to deprecate/remove for actor as well.

**Review by jianoaix — APPROVED** [review]
Looks good to fix even if we will deprecate it, since it's inaccurate right now

**Review by jjyao — APPROVED** [review]

**Comment by wuisawesome** [comment]
Link check failure looks unrelated, didn't change any code so the other tests look fine too. Merging.

**Pull request merged**